tool/live-monitor.js

```diff
@@ -0,0 +1,166 @@
+        } else if(command.startsWith('--checkeverymillis')) {
+            commands.checkEveryMillis = Number(command.slice(command.indexOf('=') + 1));
+        } else {
+            throw new Error(`Command '${command}' not recognized`);
+        }
+        return commands;
+    }, defaultCommandsValues);
+    return commands;
+};
+
+const PEGOUT_METHOD_SIGNATURES = {
+    releaseBtc: '0x',
+    updateCollections: '0x0c5a9990',
+    addSignature: '0xf10b9c59'
+};
+
+const PEGIN_METHOD_SIGNATURES = {
+    registerBtcCoinbaseTransaction: '0xccf417ae',
+    registerBtcTransaction: '0x43dc0656',
+};
+
+function isAPegoutRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGOUT_METHOD_SIGNATURES.releaseBtc:
+        case PEGOUT_METHOD_SIGNATURES.updateCollections:
+        case PEGOUT_METHOD_SIGNATURES.addSignature:
+            return true;
+        default:
+            return false;
+    }
+}
+
+function isAPeginRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGIN_METHOD_SIGNATURES.registerBtcTransaction:
+            return true;
+        default:
+            return false;
+    }
+}
+
+const commands = getParsedCommands();
+
+const web3Client = new Web3(networkParser(commands.network));
+
+let currentBlockNumber = commands.fromBlock;
+
+let notified = false;
+
+async function monitor() {
+
+    try {
+
+        const latestBlockNumber = await web3Client.eth.getBlockNumber();
+
+        if(currentBlockNumber === 'latest') {
+            currentBlockNumber = latestBlockNumber;
+        }
+
+        if(latestBlockNumber < currentBlockNumber) {
+            if(!notified) {
+                console.warn("latestBlockNumber was reached!");
+                notified = true;
+            }
+            return;
+        }
+
+        notified = false;
+
+        const block = await web3Client.eth.getBlock(currentBlockNumber, true);
+
+        console.info("Checking block: ", block.number);
+
+        currentBlockNumber++;
+
+        for(const transaction of block.transactions) {
+            
+            if(transaction.to === Bridge.address) {
+
+                // Requested only pegouts, if tx is not a pegout then continue
+                if(!commands.pegin && commands.pegout && !isAPegoutRelatedTransactionData(transaction.input)) {
+                    continue;
+                // Requested only pegins, if tx is not a pegin then return
+                } else if(!commands.pegout && commands.pegin && !isAPeginRelatedTransactionData(transaction.input)) {
+                    continue;
+                // Requested only pegins and pegouts, if tx is not a pegin or pegout then return
+                } else if(commands.pegin && commands.pegout && (!isAPeginRelatedTransactionData(transaction.input) || !isAPegoutRelatedTransactionData(transaction.input))) {
+                    continue;
+                }
```

Comment:
```suggestion
                if((commands.pegin && !isPeginRelated) || (commands.pegouts && !isPegoutRelated) {
                continue;
                }
```
That should do it, right?

Reply:
Updated!